app/Controllers/Http/UserController.js

```diff
@@ -381,6 +402,195 @@ class UserController {
       'Your account has been imported. You can now use your Franz/Ferdi account in Ferdium.',
     );
   }
+
+  async importFerdi({ request, response, view }) {
+    if (Env.get('IS_REGISTRATION_ENABLED') == 'false') {
+      // eslint-disable-line eqeqeq
+      return response.status(401).send({
+        message: 'Registration is disabled on this server',
+        status: 401,
+      });
+    }
+
+    // Validate user input
+    const validation = await validateAll(request.all(), {
+      email: 'required|email|unique:users,email',
+      password: 'required',
+    });
+    if (validation.fails()) {
+      let errorMessage =
+        'There was an error while trying to import your account:\n';
+      for (const message of validation.messages()) {
+        if (message.validation === 'required') {
+          errorMessage += `- Please make sure to supply your ${message.field}\n`;
+        } else if (message.validation === 'unique') {
+          errorMessage += '- There is already a user with this email.\n';
+        } else {
+          errorMessage += `${message.message}\n`;
+        }
+      }
+      return view.render('others.message', {
+        heading: 'Error while importing',
+        text: errorMessage,
+      });
+    }
+
+    const { email, password } = request.all();
+
+    const hashedPassword = crypto
+      .createHash('sha256')
+      .update(password)
+      .digest('base64');
+
+    if (Env.get('CONNECT_WITH_FRANZ') == 'false') {
+      // eslint-disable-line eqeqeq
+      await User.create({
+        email,
+        password: hashedPassword,
+        username: 'Franz',
+        lastname: 'Franz',
+      });
+
+      return response.send(
+        "Your account has been created but due to this server's configuration, we could not import your Ferdi account data.\n\nIf you are the server owner, please set CONNECT_WITH_FRANZ to true to enable account imports.",
+      );
+    }
+
+    const base = 'https://api.getferdi.com/';
+    const userAgent =
+      'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Franz/5.3.0-beta.1 Chrome/69.0.3497.128 Electron/4.2.4 Safari/537.36';
+
+    // Try to get an authentication token
+    let token;
+    try {
+      const basicToken = btoa(`${email}:${hashedPassword}`);
+      const loginBody = {
+        isZendeskLogin: false,
+      };
+
+      const rawResponse = await fetch(`${base}user/login`, {
+        method: 'POST',
+        body: JSON.stringify(loginBody),
+        headers: {
+          Authorization: `Basic ${basicToken}`,
+          'User-Agent': userAgent,
+          'Content-Type': 'application/json',
+          accept: '*/*',
+          'x-franz-source': 'Web',
+        },
+      });
+      const content = await rawResponse.json();
```

Comment:
This is probably the problem - you are parsing the response to JSON here

Reply:
Yeah, I think that has to be changed so this can succeed. Though, the problem seems to be in the Fetch call